### ISSUE TITLE
Dependencies break nodejs https.request

### ISSUE THREAD

**Issue by aviadbd** [mixpanel/mixpanel-node#147]
Seems like mixpanel has a dependency in https-proxy-agent, which in turns is dependant on agent-base, which replaces https.request which causes, for some reason, the strange output of:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type Function. Received type object
    at ClientRequest.once (events.js:288:11)
    at new ClientRequest (_http_client.js:151:10)
    at Object.request (https.js:281:10)
    at Object.request (/Users/aviadbendov/Dev/web/pure/node_modules/agent-base/patch-core.js:23:20)
```

for the following completely legal call (since Node 9...) - `https.request(url, options, callback)`

**Comment by Gyran** [comment]
Using agent-base less than 5 is also breaking `got` https://github.com/npm/make-fetch-happen/issues/17. 

Used https://www.npmjs.com/package/npm-force-resolutions to force `http-proxy-agent` to `5.0.0` and library works fine. Haven't tested the proxy part but the api is the same so should probably work to update `http-proxy-agent` in this lib as well.

**Comment by tdumitrescu** [comment]
Released in v0.12.0, thanks!
